.github/workflows/build_and_test_pr.yml

```diff
@@ -0,0 +1,195 @@
+name: Pull Request Testing
+
+on:
+  pull_request:
+    branches: [ develop ]  # run this on any PR pointing to develop
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  FC: gfortran-13
+  Python_REQUIRED_VERSION: 3.12.3  # 3.12.2 not available on Ubuntu 24 GHA
+  shell: bash
+
+jobs:
+  build_and_test:
+    runs-on: ${{ matrix.os }}
+    permissions:
+      pull-requests: write
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - os: macos-12
+          macos_dev_target: 12.1
+          arch: x86_64
+          python-arch: x64
+          generator: "Unix Makefiles"
+          nproc: 3
+          targets: energyplus ExpandObjects ReadVarsESO Slab Basement AppGPostProcess ParametricPreprocessor
+          run_regressions: false
+        - os: macos-14
+          macos_dev_target: 13.0
+          arch: arm64
+          python-arch: arm64
+          generator: "Unix Makefiles"
+          nproc: 3
+          targets: energyplus ExpandObjects ReadVarsESO Slab Basement AppGPostProcess ParametricPreprocessor
+          run_regressions: true
+        - os: ubuntu-24.04
+          arch: x86_64
+          python-arch: x64
+          generator: "Unix Makefiles"
+          nproc: 4
+          targets: energyplus ExpandObjects ReadVarsESO Slab Basement AppGPostProcess ParametricPreprocessor
+          run_regressions: true
+        - os: windows-2019
+          arch: x86_64
+          python-arch: x64
+          generator: "Visual Studio 16 2019"
+          nproc: 4
+          targets: energyplus ExpandObjects_build ReadVars_build Slab_build Basement_build AppGPostProcess_build ParametricPreprocessor_build
+          run_regressions: false
+        - os: windows-2022
+          arch: x86_64
+          python-arch: x64
+          generator: "Visual Studio 17 2022"
+          nproc: 4
+          targets: energyplus ExpandObjects_build ReadVars_build Slab_build Basement_build AppGPostProcess_build ParametricPreprocessor_build
+          run_regressions: true
+
+    steps:
+
+    - name: Set up Python ${{ env.Python_REQUIRED_VERSION }}
+      id: setup-python
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ env.Python_REQUIRED_VERSION }}
+
+    - name: Install Dependencies for Mac
+      if: ${{ runner.os == 'macOS' }}
+      run: |
+        brew update
+        brew install gcc@13
+        echo "FC=$(brew --prefix gcc@13)/bin/gfortran-13" >> $GITHUB_ENV
+        echo MACOSX_DEPLOYMENT_TARGET=${{ matrix.macos_dev_target }} >> $GITHUB_ENV
+
+    - name: Install Dependencies for Linux
+      if: ${{ runner.os == 'Linux' }}
+      run: |
+        sudo apt-get update
+        sudo apt-get install libxkbcommon-x11-0 xorg-dev libgl1-mesa-dev
+        if [[ "${{ matrix.os }}" == "ubuntu-24.04" ]]; then
+          # https://github.com/actions/runner-images/issues/10025
+          echo "FC=gfortran-13" >> $GITHUB_ENV
+        fi
+
+    # BUILD AND TEST INTEGRATION FILES ON THE BASELINE BRANCH
```

Comment:
For build types that we do regressions, this section checks out develop, builds it, and runs integrations.  It is possible we try to cache the results of this, I'm open to suggestion.